Checkout/Model/Quote/GetQuoteInventoryData.php

```diff
@@ -138,15 +148,39 @@ private function buildErrorResponse(string $error): ResultInterface
     private function getSalableQty(CartItemInterface $item): float
     {
         $getProductSalableQty = $this->getProductSalableQtyService();
+        $stockResolver = $this->getStockResolverService();
         try {
-            return $getProductSalableQty
-                ? $getProductSalableQty->execute($item['product']['sku'], $item->getStoreId())
+            return ($getProductSalableQty && $stockResolver)
+                ? $this->getSalableQuantity($item)
                 : $item->getProduct()->getExtensionAttributes()->getStockItem()->getQty();
         } catch (Exception $e) {
             return 0;
         }
     }
 
+    /**
+     * Get product salable qty.
+     *
+     * @param CartItemInterface $item
+     * @return float
+     * @throws LocalizedException
+     * @throws \Magento\Framework\Exception\InputException
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    private function getSalableQuantity(CartItemInterface $item): float
+    {
+        $getProductSalableQty = $this->getProductSalableQtyService();
+        $stockResolver = $this->getStockResolverService();
+        if (!$getProductSalableQty || !$stockResolver) {
+            return 0.0;
```

Comment:
You have already created $getProductSalableQty and $stockResolver. Why this redundant call and condition?